web/src/components/Header/HeaderLeftPanel.tsx

```diff
@@ -112,6 +118,34 @@ const HeaderLeftPanel: React.FC<HeaderLeftPanelProps> = ({
   useOnClickOutside(ref, () => setOpenEntryPointPicker(false))
   const [openEntryPointPicker, setOpenEntryPointPicker] = useState(false)
 
+  const [numOpsToFetch, setNumOpsToFetch] = useState(0)
+
+  // display syncing indicator when numOpsToFetch > 0
+  useEffect(() => {
+    getAdminWs().then((adminWs) => {
+      getAppWs().then((appWs) => {
+        adminWs.listDnas().then((dnas) => {
+          appWs
+            .networkInfo({
+              agent_pub_key: agentAddress as any,
+              dnas: dnas as any,
+            })
+            .then((networkInfo) => {
+              setInterval(() => {
+                let sum = 0
+                networkInfo.forEach((infoEntry) => {
+                  sum += infoEntry.fetch_pool_info.num_ops_to_fetch
+                })
+
+                setNumOpsToFetch(sum)
+              }, 5000)
+            })
+            .catch(console.error)
+        })
+      })
+    })
+  }, [])
```

Comment:
not because this component will necessarily ever get unmounted, but take a look into methods for unmounting a component (lifecycle handling) and tearing down a setTimeout or setInterval again